spring-integration-core/src/main/java/org/springframework/integration/support/json/MutableMessageJacksonDeserializer.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright 2017 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.integration.support.json;
+
+import java.io.IOException;
+
+import org.springframework.integration.support.MutableMessage;
+import org.springframework.integration.support.MutableMessageHeaders;
+import org.springframework.util.ClassUtils;
+
+import com.fasterxml.jackson.databind.DeserializationContext;
+import com.fasterxml.jackson.databind.JsonNode;
+
+/**
+ * The {@link MessageJacksonDeserializer} implementation for the {@link MutableMessage}.
+ *
+ * @author Artem Bilan
+ *
+ * @since 5.0
+ */
+public class MutableMessageJacksonDeserializer extends MessageJacksonDeserializer<MutableMessage<?>> {
```

Comment:
I use `MutableMessage` there in the  code like:
```
if (this.canProcessMessageList) {
	message = new MutableMessage<>(parameters.getMessages(), parameters.getHeaders());
}
```
because `MutableMessageHeaders` doesn't generate `ID` and `TIMESTAMP`.
I need the message object there anyway because of `CollectionArgumentResolver` and as you see restricted contract of the `HandlerMethodArgumentResolver.resolveArgument()` where we can have only one `message` to consult.